jpype/_jarray.py

```diff
@@ -15,14 +15,34 @@
 #   See NOTICE file for details.
 #
 # *****************************************************************************
+import abc
+import collections.abc
 import _jpype
+import typing
 from . import _jcustomizer
 
 
 __all__ = ['JArray']
 
 
-class JArray(_jpype._JObject, internal=True):  # type: ignore[call-arg]
+T = typing.TypeVar('T')
+
+
+class _JArrayGeneric(collections.abc.Sequence[T]):
```

Comment:
Note that this could live in a `pyi` file (like https://github.com/jpype-project/jpype/blob/master/jpype/_jcollection.pyi).

I can definitely see the advantage of it being in the same file though - whatever we choose, we should be consistent.